.github/workflows/ci.yml

```diff
@@ -46,16 +46,21 @@ jobs:
     name: Publish release
     runs-on: ubuntu-latest
     needs: tests
+    environment: release
+    permissions:
+      id-token: write  # Needed for trusted publishing
 
     if: ${{ startsWith(github.ref, 'refs/tags/') }}
 
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
-          python-version: "3.10"
+          python-version: "3.12"
 
-      - run: pipx run nox -s release
-        env:
-          FLIT_USERNAME: __token__
-          FLIT_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+      - run: |
+          pip install build
+          python -m build
```

Comment:
@takluyver this bit is problematic — it's discouraged to build in the same job that has elevated privileges and is able to access OIDC.

I recommend splitting it out due to security concerns: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/